app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt

```diff
@@ -1,46 +1,79 @@
-        }
-        binding.button1.setOnClickListener {
-            binding.textView.text ="1"
-        }
-        binding.button2.setOnClickListener {
-            binding.textView.text ="2"
-        }
-        binding.button3.setOnClickListener {
-            binding.textView.text ="3"
-        }
-        binding.button4.setOnClickListener {
-            binding.textView.text ="4"
-        }
-        binding.button5.setOnClickListener {
-            binding.textView.text ="5"
-        }
-        binding.button6.setOnClickListener {
-            binding.textView.text ="6"
-        }
-        binding.button7.setOnClickListener {
-            binding.textView.text ="7"
+        binding.button0.setOnClickListener { clickOperand(0) }
+        binding.button1.setOnClickListener { clickOperand(1) }
+        binding.button2.setOnClickListener { clickOperand(2) }
+        binding.button3.setOnClickListener { clickOperand(3) }
+        binding.button4.setOnClickListener { clickOperand(4) }
+        binding.button5.setOnClickListener { clickOperand(5) }
+        binding.button6.setOnClickListener { clickOperand(6) }
+        binding.button7.setOnClickListener { clickOperand(7) }
+        binding.button8.setOnClickListener { clickOperand(8) }
+        binding.button9.setOnClickListener { clickOperand(9) }
+
+        binding.buttonPlus.setOnClickListener { clickOperator("+") }
+        binding.buttonMinus.setOnClickListener {clickOperator("-")}
+        binding.buttonDivide.setOnClickListener {clickOperator("/")}
+        binding.buttonMultiply.setOnClickListener {clickOperator("*")}
+
+        binding.buttonDelete.setOnClickListener { clickDelete() }
+        binding.buttonEquals.setOnClickListener { clickEquals() }
+
+
+    }
+    private fun clickOperator(operator: String){
+        if(expression.isNotEmpty() && expression.last().isDigit()){
+            expression += " $operator"
         }
-        binding.button8.setOnClickListener {
-            binding.textView.text ="8"
+        binding.textView.text = expression
+    }
+
+    private fun clickOperand(value: Int){
+        if(expression.isNotEmpty() && !expression.last().isDigit()){
+            expression += " "
         }
-        binding.button9.setOnClickListener {
-            binding.textView.text ="9"
+        expression += value.toString()
+        binding.textView.text = expression
+    }
+
+    private fun clickDelete(){
+        if(expression.isEmpty()) return
+        // 앞의문자가 공백이거나 소수점이면 같이 지워버리기
+        val deleteSize =
+            if (expression.length > 1 && (expression[expression.lastIndex - 1] == ' ' || expression[expression.lastIndex - 1] == '.')) 2 else 1
+        expression = expression.substring(0, expression.length - deleteSize)
+
+        binding.textView.text = expression
+    }
+
+    private fun clickEquals(){
+        if(!expression.last().isDigit()){
+            Toast.makeText(this, WRONG_EXPRESSION, Toast.LENGTH_SHORT).show()
+        }else{
```

Comment:
```suggestion
        if (!expression.last().isDigit()) {
            Toast.makeText(this, WRONG_EXPRESSION, Toast.LENGTH_SHORT).show()
        } else {
```
포메팅이 안되어 있는 것 같네요.
포메팅을 해보면 어떨까요?

Reply:
앗 실수로 설정에서 자동포맷팅을 안해놨군요 ㅠㅜ